### PR TITLE
os/fs/tmpfs: Fix define memory alloc function

### PR DESCRIPTION
**Pull request by KimDongEon**
Redefine memory alloc function when using multi heap at tmpfs header file

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [80954454a99b1a6e95ed77c971096be3ef12c6b6] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [80954454a99b1a6e95ed77c971096be3ef12c6b6] - Code Rule Check (C++) OK.

File: os/fs/tmpfs/fs_tmpfs.h [hunk 1]
```diff
@@ -75,6 +75,17 @@
 
 #define TFO_FLAG_UNLINKED (1 << 0)  /* Bit 0: File is unlinked */
 
+/* Redefine memory alloc function when using multi heap */
+
+#if CONFIG_MM_NHEAPS > 1 && CONFIG_MM_REGIONS > 1
+#undef		kmm_malloc
```
Comment (drashti304) [inline os/fs/tmpfs/fs_tmpfs.h]: Please correct the formatting

Comment (sunghan-chang) [reply os/fs/tmpfs/fs_tmpfs.h]: Hello @drashti304 .
Could you let me know what kind formatting do you mean?

Comment (drashti304) [reply os/fs/tmpfs/fs_tmpfs.h]: I think the tab after #undef is not required.
in file view it looks like below:
#undef         kmm_realloc
#define  kmm_malloc(s)

Comment (drashti304) [reply os/fs/tmpfs/fs_tmpfs.h]: I think the tab after #undef is not required.
in file view it looks like below:
#undef \t kmm_realloc
#define kmm_malloc(s) malloc_at(CONFIG_FS_TMPFS_HEAP_INDEX, s)

Comment (drashti304) [reply os/fs/tmpfs/fs_tmpfs.h]: I think the tab after #undef is not required.
in file view it looks like below:
formatting

Comment (KimDongEon) [reply os/fs/tmpfs/fs_tmpfs.h]: I will modify

**Comment by seinfra** [comment]
Target : [51e44ce44ef5301710018a5ed2f6b4edc2baa693] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [51e44ce44ef5301710018a5ed2f6b4edc2baa693] - Code Rule Check (C++) OK.

File: os/fs/tmpfs/fs_tmpfs.h [hunk 2]
```diff
@@ -75,6 +75,17 @@
...
 
+/* Redefine memory alloc function when using multi heap */
+
+#if CONFIG_MM_NHEAPS > 1 && CONFIG_MM_REGIONS > 1
+#undef kmm_malloc
+#undef kmm_zalloc
+#undef kmm_realloc
+#define	kmm_malloc(s)		malloc_at(CONFIG_FS_TMPFS_HEAP_INDEX, s)
+#define	kmm_zalloc(s)		zalloc_at(CONFIG_FS_TMPFS_HEAP_INDEX, s)
+#define	kmm_realloc(p, s)	realloc_at(CONFIG_FS_TMPFS_HEAP_INDEX, p, s)
```
Comment (sunghan-chang) [inline os/fs/tmpfs/fs_tmpfs.h]: Please modify here as well

Comment (KimDongEon) [reply os/fs/tmpfs/fs_tmpfs.h]: I modify

**Comment by seinfra** [comment]
Target : [3fff4b635b811a8f0179b2ef829a24771f2a347c] -

**Comment by seinfra** [comment]
Target : [3fff4b635b811a8f0179b2ef829a24771f2a347c] -

**Comment by seinfra** [comment]
Target : [d43df6ba86319e5a568254fb459b282e52fe28ae] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [d43df6ba86319e5a568254fb459b282e52fe28ae] - Code Rule Check (C++) OK.

**Pull request merged**